tsl/src/fdw/data_node_scan_exec.c

```diff
@@ -71,22 +71,13 @@ data_node_scan_next(CustomScanState *node)
 	oldcontext = MemoryContextSwitchTo(econtext->ecxt_per_tuple_memory);
 	slot = fdw_scan_iterate(&node->ss, &sss->fsstate);
 	MemoryContextSwitchTo(oldcontext);
-	/*
-	 * If any system columns are requested, we have to force the tuple into
-	 * physical-tuple form to avoid "cannot extract system attribute from
-	 * virtual tuple" errors later.  We also insert a valid value for
-	 * tableoid, which is the only actually-useful system column.
-	 */
-	if (sss->systemcol && !TupIsNull(slot))
-	{
-#if PG12_LT
-		HeapTuple tup = ExecMaterializeSlot(slot);
 
-		tup->t_tableOid = RelationGetRelid(node->ss.ss_currentRelation);
-#else
-		slot->tts_tableOid = RelationGetRelid(node->ss.ss_currentRelation);
-#endif
-	}
+	/* Raise an error when system column is requsted, eg. tableoid */
+	if (sss->systemcol && !TupIsNull(slot))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("system columns are not accessible on distributed hypertable "
+						"with enabled timescaledb.enable_per_data_node_queries")));
```

Comment:
Main error should be short according to error style guide. I suggest splitting this up:

```suggestion
		ereport(ERROR,
				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
				 errmsg("system columns are not accessible on distributed hypertables"),
				 errhint("Set timescaledb.enable_per_data_node_queries=false to query system columns.")));
```

Reply:
> "system columns are not accessible on distributed hypertables"

This suggested text is not fully correct, since system columns are accessible on distributed hypetable if the optimization is disabled. So I don't think it should be stated in the suggested way.

Reply:
Updated the message to be more specific